### PR TITLE
K8SPXC-899: Add HAProxy domain names to ssl-internal certificate

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPXC-899](https://jira.percona.com/browse/K8SPXC-899) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPXC-899

Also add pxc domain name from CN to the ssl-internal certificate, because running with `--ssl-mode=VERIFY_IDENTITY` doesn't work without it

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| haproxy | passed |
| recreate | passed |
| scaling | passed |
| storage | passed |
| big-data | passed |
| scheduled-backup | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| self-healing-chaos | failed |
| init-deploy | failed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| demand-backup | passed |
| smart-update | passed |
| limits | passed |
| monitoring-2-0 | passed |
| pitr | passed |
| affinity | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | failed |
| demand-backup-encrypted-with-tls | failed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1074/commits/65f45740e1f5096b87df70086d143f16aef938b4
image: `perconalab/percona-xtradb-cluster-operator:PR-1074-65f45740`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| upgrade-haproxy | passed |
| recreate | passed |
| scheduled-backup | passed |
| storage | passed |
| haproxy | passed |
| big-data | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| limits | passed |
| demand-backup | passed |
| smart-update | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| upgrade-consistency | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1074/commits/6e4949112ec236ad411ccd2a6542b3b8f79c34a0
image: `perconalab/percona-xtradb-cluster-operator:PR-1074-6e494911`

File: e2e-tests/tls-issue-cert-manager/conf/some-name-tls-issue.yml [hunk 1]
```diff
@@ -28,22 +28,17 @@ spec:
             storage: 2Gi
     affinity:
       antiAffinityTopologyKey: "kubernetes.io/hostname"
-  proxysql:
+  haproxy:
```
Comment (hors) [inline e2e-tests/tls-issue-cert-manager/conf/some-name-tls-issue.yml:31]: I do not think that we need to remove proxysql we need to add additional test case

File: pkg/controller/pxc/tls.go [hunk 2]
```diff
@@ -149,7 +149,14 @@ func (r *ReconcilePerconaXtraDBCluster) createSSLByCertManager(cr *api.PerconaXt
...
 			CommonName: cr.Name + "-pxc",
 			DNSNames: []string{
+				cr.Name + "-pxc",
 				"*." + cr.Name + "-pxc",
+				cr.Name + "-haproxy-replicas." + cr.Namespace + ".svc.cluster.local",
+				cr.Name + "-haproxy-replicas." + cr.Namespace,
+				cr.Name + "-haproxy-replicas",
+				cr.Name + "-haproxy." + cr.Namespace + ".svc.cluster.local",
+				cr.Name + "-haproxy." + cr.Namespace,
+				cr.Name + "-haproxy",
```
Comment (hors) [inline pkg/controller/pxc/tls.go:159]: should we add `-haproxy-replicas` if the service was disabled or should we add haproxy at all if we have proxysql only 
@delgod  what do you think ? I think no...

**Review by hors — CHANGES_REQUESTED** [review]

File: pkg/controller/pxc/tls.go (hunk 2 above)
Comment (delgod) [reply pkg/controller/pxc/tls.go:159]: I think we should add haproxy always, reason: user can change options anytime after

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| storage | passed |
| big-data | passed |
| scaling | passed |
| haproxy | passed |
| recreate | passed |
| scheduled-backup | failed |
| self-healing | passed |
| self-healing-chaos | passed |
| scaling-proxysql | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| smart-update | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| upgrade-consistency | failed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1074/commits/8534571ee6cf0ed23f490c8dfec40265f00ee02c
image: `perconalab/percona-xtradb-cluster-operator:PR-1074-8534571e`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| recreate | passed |
| big-data | passed |
| haproxy | passed |
| upgrade-haproxy | passed |
| scheduled-backup | passed |
| restore-to-encrypted-cluster | passed |
| init-deploy | passed |
| demand-backup | passed |
| limits | passed |
| pitr | passed |
| monitoring-2-0 | passed |
| demand-backup-encrypted-with-tls | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| scaling | passed |
| tls-issue-cert-manager | passed |
| scaling-proxysql | passed |
| tls-issue-cert-manager-ref | passed |
| security-context | passed |
| validation-hook | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| self-healing-advanced | passed |
| self-healing-advanced-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1074/commits/8534571ee6cf0ed23f490c8dfec40265f00ee02c
image: `perconalab/percona-xtradb-cluster-operator:PR-1074-8534571e`

File: e2e-tests/tls-issue-cert-manager/conf/some-name-tls-issue.yml (hunk 1 above)
Comment (pooknull) [reply e2e-tests/tls-issue-cert-manager/conf/some-name-tls-issue.yml]: @hors ProxySQL returned in 8534571ee6cf0ed23f490c8dfec40265f00ee02c

**Review by delgod — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**